frontend/src/services/parsing/JsonParser.js

```diff
@@ -11,6 +11,19 @@ const JsonParser = {
 
       return result;
    }
+
+   static parse( text ) {
+      let result;
+
+      try {
+         result = JSON.parse( text );
+      }
+      catch ( e ) {
+         throw new Error( `Not JSON: ${text}` );
+      }
+
+      return result;
+   }
```

Comment:
```js
function parse(text) {
  try {
    return JSON.parse(text);
  } catch(e) {
    throw new Error( `Not JSON: ${text}` );
  }
}
```
This would be much simpler. Also introducing `let` without modifying it isn't a good practice. Either use `const` or return value without assigning it to a variable.

Reply:
Thanks. Fixed in #52 .